helix-core/src/main/java/org/apache/helix/task/TaskDriver.java

```diff
@@ -388,11 +389,19 @@ public void enqueueJobs(final String queue, final List<String> jobs,
       // if queue is full, Helix will try to clean up the expired job to free more space.
       WorkflowContext workflowContext = TaskUtil.getWorkflowContext(_propertyStore, queue);
       if (workflowContext != null) {
-        Set<String> expiredJobs =
-            TaskUtil.getExpiredJobs(_accessor, _propertyStore, workflowConfig, workflowContext);
+        Set<String> expiredJobs = Sets.newHashSet();
+        Set<String> jobsWithoutConfig = Sets.newHashSet();
+        TaskUtil.getExpiredJobs(_accessor, _propertyStore, workflowConfig, workflowContext,
+            expiredJobs, jobsWithoutConfig);
         if (!TaskUtil.removeJobsFromWorkflow(_accessor, _propertyStore, queue, expiredJobs, true)) {
           LOG.warn("Failed to clean up expired and completed jobs from queue " + queue);
         }
+
+        if (!TaskUtil.removeJobsFromWorkflow(_accessor, _propertyStore, queue, jobsWithoutConfig,
+            true)) {
+          LOG.warn("Failed to clean up jobsWithoutConfig jobs from queue {}" , queue);
```

Comment:
Fix the log message:

Failed to clean up jobs without JobConfigs from queue's DAG. Queue name: ...

Note: I find myself commenting on stuff like this often - let's try to have a good, succinct log message before submitting the diff for review :)

Reply:
Fixed. Thanks for the comment.